src/Control/Monad/ST.purs

```diff
@@ -62,3 +62,8 @@ foreign import runST
 -- | parentheses instead.
 pureST :: forall a. (forall h. Eff (st :: ST h) a) -> a
 pureST st = runPure (runST st)
+
+-- | Safely coerce a locally mutable `STRef h a` to a globally mutable `Ref a`
+-- | discarding the phantom type `h`.
+toRef :: forall a h. (STRef h a) -> (Ref a)
```

Comment:
Actually, we shouldn't provide this function. It's unsafe, since a pure value could be computed using the `STRef`, but then modified later using the `Ref`. The type system won't prevent the `Ref` from escaping here:

```purescript
danger = pureST do
  ref <- newSTRef 0
  ...
  final <- readSTRef ref
  pure { value: final, ref :: toRef ref }
```